sql/2020/03_Markup/pages_almanac_by_device_and_html_lang.sql

```diff
@@ -0,0 +1,44 @@
+#standardSQL
+# page almanac metrics grouped by device and html lang
+
+# helper to create percent fields
+CREATE TEMP FUNCTION AS_PERCENT (freq FLOAT64, total FLOAT64) RETURNS FLOAT64 AS (
+  ROUND(SAFE_DIVIDE(freq, total), 4)
+);
+
+# returns all the data we need from _almanac
+CREATE TEMPORARY FUNCTION get_almanac_html_lang(almanac_string STRING)
+RETURNS STRING LANGUAGE js AS '''
+try {
+    var almanac = JSON.parse(almanac_string); 
+
+    if (Array.isArray(almanac) || typeof almanac != 'object') return '';
+
+    if (almanac.html_node && almanac.html_node.lang) {
+      return almanac.html_node.lang.trim().toLowerCase();
```

Comment:
Does this include region codes, eg `en-US`, or just language codes? Do you need to parse out the language codes?

Reply:
It's just the raw values trimmed and in lowercase. I've already done a live query on this. Let's see if it's enough for the team. If not I can invent something new. It includes frequency data so it should be processable in the sheet.